go/pvl/interp.go

```diff
@@ -918,3 +926,59 @@ func pvlSubstitute(template string, state PvlScriptState, match []string) (strin
 	}
 	return res, nil
 }
+
+// Take an instruction and if it specifies a custom error via an "error" key, replace the error.
+// Always returns an error because that's its job. The second return argument is true if a different error is returned.
+// If there is an issue with the "error" spec, this just returns the unmodfied err1.
+// It would be just too harsh to report INVALID_PVL for that.
+func pvlCustomError(g ProofContextExt, ins *jsonw.Wrapper, state PvlScriptState, err1 libkb.ProofError) (libkb.ProofError, bool) {
+	if err1 == nil {
+		return err1, false
+	}
+	if !pvlJSONHasKey(ins, "error") {
+		return err1, false
+	}
+
+	var ename string
+	edesc := err1.GetDesc()
+
+	ar, err := ins.AtKey("error").ToArray()
+	if err != nil {
+		// "error": "name"
+		name, err := ins.AtKey("error").GetString()
+		if err != nil {
+			debugWithState(g, state, "Invalid error spec: %v", err)
+		} else {
+			ename = name
+		}
+	} else {
+		// "error": ["name", "desc"]
+		name, err := ar.AtIndex(0).GetString()
+		if err != nil {
+			debugWithState(g, state, "Invalid error spec: %v", err)
```

Comment:
the problem with early-out here is this that if someone bungles the error like this
```
"error": [220, "Missing sobriquet"]
```
which is invalid because 220 is a number not a string like "BAD_API_URL"